src/index.ts

```diff
@@ -416,24 +416,39 @@ const ruleFunction: StylelintPlugin<PrimaryOptions, SecondaryOptions> =
 
           // support auto fixing
           if (context.fix && !disableFix && autoFixFuncNormalized) {
-            const fixedValue = autoFixFuncNormalized(
-              node,
-              {
-                validVar,
-                validFunc,
-                validKeyword,
-                validValue,
-                longhandProp,
-                longhandValue,
-              },
-              root,
-              config
-            );
-
-            // apply fixed value if returned
-            if (fixedValue) {
-              // eslint-disable-next-line no-param-reassign
-              node.value = fixedValue;
+            try {
+              const fixedValue = autoFixFuncNormalized(
+                node,
+                {
+                  validVar,
+                  validFunc,
+                  validKeyword,
+                  validValue,
+                  longhandProp,
+                  longhandValue,
+                },
+                root,
+                config
+              );
+
+              // apply fixed value if returned
+              if (fixedValue) {
+                // eslint-disable-next-line no-param-reassign
+                node.value = fixedValue;
+              }
+            } catch (err: unknown) {
+              if (err instanceof Error) {
+                utils.report({
+                  ruleName,
+                  result,
+                  node,
+                  line: node.source!.start?.line,
+                  message: err.toString(),
```

Comment:
Please use [`stylelint.utils.ruleMessages`](https://stylelint.io/developer-guide/plugins/#stylelintutilsrulemessages).

And here are examples of how they are used in this plugin:
- https://github.com/AndyOGo/stylelint-declaration-strict-value/blob/master/src/index.ts#L25-L27
- https://github.com/AndyOGo/stylelint-declaration-strict-value/blob/master/src/lib/validation.ts#L187-L227
- https://github.com/AndyOGo/stylelint-declaration-strict-value/blob/master/src/index.ts#L443-L450